rsconf/package_data/docker/main.sh.jinja

```diff
@@ -28,23 +40,45 @@ Then re-run rsconf
         dnf -y install dnf-plugins-core
         dnf config-manager \
             --add-repo \
-            https://download.docker.com/linux/fedora/docker-ce.repo
+            https://download.docker.com/linux/fedora/"$_docker_pkg".repo
     else
         yum-config-manager \
-            --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+            --add-repo https://download.docker.com/linux/centos/"$_docker_pkg".repo
         yum makecache fast
         # yum-plugin-ovl required for overlay2 on centos7
         # https://github.com/docker-library/official-images/issues/1291
         rsconf_yum_install yum-plugin-ovl
     fi
-#TODO(robnagler) how to update?
-    rsconf_yum_install docker-ce
+    rsconf_yum_install "$_docker_pkg"
     {% if rsconf_db.channel == "dev" %}
         # Give vagrant user access in dev mode only
         usermod -aG docker vagrant
     {% endif %}
+    if docker_should_update; then
+        install_err "Installed docker=$(docker version) is < min_version={{ docker.min_version }} after install."
+    fi
 }
 
 docker_main() {
     docker_install
 }
+
+docker_should_update() {
+    docker_should_update_helper Client || docker_should_update_helper Server
+}
+
+docker_should_update_helper() {
```

Comment:
We haven't use "helper" in the past. I would usually call this docker_should_update_one or docker_should_update_do or even docker_should_update_component.

Reply:
0349609